Bugsnag/KSCrash/Source/KSCrash/Recording/BSG_KSSystemInfo.m

```diff
@@ -313,96 +301,19 @@ + (NSString *)currentCPUArch {
     return result ?: [NSString stringWithUTF8String:bsg_ksmachcurrentCPUArch()];
 }
 
-/** Check if the current device is jailbroken.
- *
- * @return YES if the device is jailbroken.
- */
-+ (BOOL)isJailbroken {
-    return is_jailbroken();
-}
-
-/** Check if the current build is a debug build.
- *
- * @return YES if the app was built in debug mode.
- */
-+ (BOOL)isDebugBuild {
-#ifdef DEBUG
-    return YES;
-#else
-    return NO;
-#endif
-}
-
-/** Check if this code is built for the simulator.
- *
- * @return YES if this is a simulator build.
- */
-+ (BOOL)isSimulatorBuild {
-#if BSG_PLATFORM_SIMULATOR
-    return YES;
-#else
-    return NO;
-#endif
-}
-
-/** The file path for the bundle’s App Store receipt.
- *
- * @return App Store receipt for iOS 7+, nil otherwise.
- */
-+ (NSString *)receiptUrlPath {
-    return [NSBundle mainBundle].appStoreReceiptURL.path;
-}
-
-/** Check if the current build is a "testing" build.
- * This is useful for checking if the app was released through Testflight.
- *
- * @return YES if this is a testing build.
- */
-+ (BOOL)isTestBuild {
-    return [[self receiptUrlPath].lastPathComponent
-        isEqualToString:@"sandboxReceipt"];
-}
-
-/** Check if the app has an app store receipt.
- * Only apps released through the app store will have a receipt.
- *
- * @return YES if there is an app store receipt.
- */
-+ (BOOL)hasAppStoreReceipt {
-    NSString *receiptPath = [self receiptUrlPath];
-    if (receiptPath == nil) {
-        return NO;
-    }
-    BOOL isAppStoreReceipt =
-        [receiptPath.lastPathComponent isEqualToString:@"receipt"];
-    BOOL receiptExists =
-        [[NSFileManager defaultManager] fileExistsAtPath:receiptPath];
-
-    return isAppStoreReceipt && receiptExists;
-}
-
-+ (NSString *)buildType {
-    if ([BSG_KSSystemInfo isSimulatorBuild]) {
-        return @"simulator";
-    }
-    if ([BSG_KSSystemInfo isDebugBuild]) {
-        return @"debug";
-    }
-    if ([BSG_KSSystemInfo isTestBuild]) {
-        return @"test";
-    }
-    if ([BSG_KSSystemInfo hasAppStoreReceipt]) {
-        return @"app store";
-    }
-    return @"unknown";
-}
-
 // ============================================================================
 #pragma mark - API -
 // ============================================================================
 
-+ (NSDictionary *)systemInfo {
-    NSMutableDictionary *sysInfo = [NSMutableDictionary dictionary];
+/**
+ * Returns a systemInfo dictionary containing all the nonvolatile unchanging values.
+ */
++ (NSDictionary *)systemInfoStatic {
+    static NSMutableDictionary *sysInfo;
+    if (sysInfo) {
```

Comment:
Should this `dispatch_once` to avoid weirdness in the case that the first couple of invocations happen at about the same time on different threads?

Reply:
I didn't think any weirdness was possible because the first invocation is predictable but have now wrapped it with `dispatch_once` anyway 👍